crd/nodenetworkconfig/client.go

```diff
@@ -136,3 +144,20 @@ func (c *Client) UpdateSpec(ctx context.Context, key types.NamespacedName, spec
 	}
 	return nnc, nil
 }
+
+// SetOwnerRef sets the owner of the NodeNetworkConfig to the given object, using HTTP Patch
+func (c *Client) SetOwnerRef(ctx context.Context, nnc *v1alpha.NodeNetworkConfig, owner metav1.Object) error {
+	newNNC := nnc.DeepCopy()
+	if err := ctrlutil.SetControllerReference(owner, newNNC, Scheme); err != nil {
+		// SetControllerReference returns an error if the object already has a different owner or
+		// if there are issues with the scheme. Since we control the scheme in this package, we can safely
+		// assume than any error is caused by an owner already existing
+		return &OwnerExistsError{}
```

Comment:
why not lean on ctrlutil's [`AlreadyOwnedError`](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/controller/controllerutil/controllerutil.go#L38)?

Reply:
because I didn't read the docs and didn't know it existed 😆